java-manta-client-unshaded/src/main/java/com/joyent/manta/http/RequestIdInterceptor.java

```diff
@@ -61,6 +61,12 @@
 
     @Override
     public void process(final HttpRequest request, final HttpContext context) throws HttpException, IOException {
+        if (request.containsHeader(MantaHttpHeaders.REQUEST_ID)) {
+            // See: MANTA-3673, manta barfs on malformed x-request-id
```

Comment:
I would log a warning and throw away the UUID and display the new UUID. Also, I would not check in the TODO.